detox/src/Detox.js

```diff
@@ -109,14 +78,19 @@ class Detox {
 
     if (this._server) {
       await this._server.close();
+      this._server = null;
     }
 
     if (argparse.getArgValue('cleanup') && this.device) {
       await this.device.shutdown();
     }
+
+    this.device = null;
   }
 
   async beforeEach(testSummary) {
+    await this._assertNoPendingInit();
```

Comment:
c&p: really wish you'd cover this logic using unit tests

Reply:
Covered with test: `Calling detox.afterEach() before detox.init() completes, throws error`